feat: change default ignore pattern to `**/node_modules/` in flat config

mdjermanovic opened the pull request:
<!--
    Thank you for contributing!

    ESLint adheres to the [JS Foundation Code of Conduct](https://eslint.org/conduct).
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

<!--
    The following template is intentionally not a markdown checkbox list for the reasons
    explained in https://github.com/eslint/eslint/pull/12848#issuecomment-580302888
-->

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[x] Add something to the core
[ ] Other, please explain:

Fixes #17113

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/latest/contribute/pull-requests)
    - Include tests for this change
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)

Changed the default ignore pattern to `**/node_modules/`, updated tests, and added an example of how to unignore a directory in node_modules.

#### Is there anything you'd like reviewers to focus on?

<!-- markdownlint-disable-file MD004 -->

fasttime commented:
Thanks for this PR! I would recommend to also update this message, as it could be slightly misleading:

https://github.com/eslint/eslint/blob/0c415cda5d76dbe5120ab9f3c4c81320538e35f0/lib/eslint/eslint-helpers.js#L601

We could also simply point users to the [documentation](https://eslint.org/docs/latest/use/configure/configuration-files-new#globally-ignoring-files-with-ignores) instead of suggesting to add CLI options (because IINM there is no guarantee that the CLI is being used here?).

mdjermanovic commented:
> Thanks for this PR! I would recommend to also update this message, as it could be slightly misleading:
> 
> https://github.com/eslint/eslint/blob/0c415cda5d76dbe5120ab9f3c4c81320538e35f0/lib/eslint/eslint-helpers.js#L601

Good catch! Updated in https://github.com/eslint/eslint/pull/17184/commits/3fa6c9243ee52dc88625be2e0ab1647ccd11da1f.

> We could also simply point users to the [documentation](https://eslint.org/docs/latest/use/configure/configuration-files-new#globally-ignoring-files-with-ignores) instead of suggesting to add CLI options (because IINM there is no guarantee that the CLI is being used here?).

Since we can't know at this point how is eslint being used and whether the pattern should be added in a config file or CLI command or perhaps in a configuration for a third-party tool, I changed "ignore-pattern" to generic "ignore pattern".

fasttime reviewed (approved):
LGTM!

mdjermanovic commented:
Rebased to fix merge conflicts.
